subprojects/ide-provisioning/build.gradle.kts

```diff
@@ -0,0 +1,29 @@
+plugins {
+    id("profiler.embedded-library")
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(17))
```

Comment:
Huh, it seems like `ide-starter` was compiled using Java 17:
`class file has wrong version 61.0, should be 52.0`

The public API of `ide-provisioning` subproject is not contains any classes from `ide-starter`, so I think it should be possible somehow to configure toolchains in such a way, to use `ide-starter` with Java 17 internally, but compile `ide-provisioning` subproject using Java 8. Do you know how to achieve that?

Reply:
The problem is not just compiling, but also running classes compiled with Java17 on JDK < 17. So then we would need gradle-profiler to run only on JDK >= 17 or at least require JDK >= 17 when doing sync profiling.

Reply:
@asodja How to do it correctly?

Reply:
Good question. I believe we still want to profile with gradle-profiler on JDK8 for most scenarions, but maybe we could soften that for sync profiling.

I guess we would need to:
1. Add a check when user tries to profile sync and they use JDK < 17 we fail
2. Add some interface to call IDE provisioning
3. Use multirelease jar or ServiceLoader to implement Java17 implementation

Reply:
I think we need to put some more thoughts in to that

Reply:
@asodja I think for now  we can check `JDK < 17` only if user enabled IDE auto-provisioning for sync. In future, when `profiler` will use `ide-starter` for sync triggering as well, we could think about it extend scope of that check, yes

Reply:
@asodja Another "possible" option is fork `ide-starter` and build it from source using 8, right? But supporting a fork can be painful tho.

Reply:
I would not fork `ide-starter`, since that means extra work to maintain it as you mentioned. And I am sure we don't have resources for that. 

Theoretically we could also just call that part of code with JDK >= 17 (so user would somehow provide a path to some JDK17 compatbile JDK), but I think having a requirement to use JDK17 for sync is ok. 

I would maybe just ask on Slack if anyone has any concern regarding that (or possible if they see any other option).